compiler/rustc_session/src/config.rs

```diff
@@ -2468,6 +2468,15 @@ pub fn build_session_options(early_dcx: &mut EarlyDiagCtxt, matches: &getopts::M
         }
     }
 
+    if !nightly_options::is_unstable_enabled(matches)
```

Comment:
Annoyingly, there seem to be multiple different ways of checking for `-Zunstable-options` when implementing these ad-hoc value checks:

- Some code uses `nightly_options::is_unstable_enabled(matches)`, which parses the value independently
- Some code uses `unstable_opts.unstable_options`, which uses whatever has already been parsed into the `UnstableOptions` struct via normal means

In this context, we should be able to just check `unstable_opts.unstable_options`, which seems like the less hacky variant.

(I assume you probably just copied this from the code immediately below. 🤷)

Reply:
I did check the unstable_options.unstable_options thing, but it says you shouldn't touch it directly, so I avoided poking it directly. idk lol?

I think there should be a different way of specifying "these values are unstable" that uses a pattern or pattern-guard of some kind.

Reply:
I realize there's only a few cases of this so it seems like a waste, but I think a profusion of flags is partly the fault of it not being easy to add such patterns that determine stability.

Reply:
I don’t want to derail this PR too much, but if I get a chance I’ll open a separate issue or thread to discuss improvements to unstable values of stable flags, because the status quo has been annoying me too.